ecommerce/frontend/src/components/product/ProductImageGallery.js

```diff
@@ -0,0 +1,80 @@
+import './ProductImageGallery.less'
+
+import { Pagination, Rate, Skeleton } from 'antd'
+import React, { useState } from 'react'
+import cls from 'classnames'
+
+export const ProductImageGallery = ({ product, loading = false }) => {
+    const [active, setActive] = useState(0)
+
+    const getImages = () => {
+        if (loading) return []
+
+        return [
+            product.image_url,
+            'https://picsum.photos/id/237/200/300',
+            'https://picsum.photos/id/102/400/500',
+            'https://picsum.photos/id/27/200/300',
+            'https://picsum.photos/id/72/600/300',
+        ]
+    }
+
+    const images = getImages()
+
+    const onDetailImageClick = ix => () => {
+        setActive(ix)
+    }
+
+    if (loading) {
+        return (
+            <div className="product-image-gallery">
+                <div className="product-image-gallery-side">
+                    <Skeleton.Avatar className="product-image-gallery-detail" shape="square" active />
+                    <Skeleton.Avatar className="product-image-gallery-detail" shape="square" active />
+                    <Skeleton.Avatar className="product-image-gallery-detail" shape="square" active />
+                    <Skeleton.Avatar className="product-image-gallery-detail" shape="square" active />
+                </div>
+                <div className="product-image-gallery-main-container">
+                    <Skeleton.Image className="product-image-gallery-main" shape="square" active />
+                </div>
+            </div>
+        )
+    }
+
+    console.log(images.length, active)
```

Comment:
There are many console.logs here and there, make sure you delete most of the unnecessary ones

Reply:
Yup, will do. Thanks